grakn-engine/src/main/java/ai/grakn/engine/controller/GraphFactoryController.java

```diff
@@ -75,27 +76,29 @@ public GraphFactoryController(Service spark) {
     @ApiImplicitParam(name = "graphConfig", value = "The type of graph config to return", required = true, dataType = "string", paramType = "path")
     private String getGraphConfig(Request request, Response response) {
         String graphConfig = request.queryParams(REST.Request.GRAPH_CONFIG_PARAM);
-        ConfigProperties prop = ConfigProperties.getInstance();
+        if(graphConfig == null){
+            graphConfig = REST.GraphConfig.DEFAULT;
+        }
 
-        try {
-            if (graphConfig == null) {
-                graphConfig = ConfigProperties.GRAPH_CONFIG_PROPERTY;
-            } else {
-                switch (graphConfig) {
-                    case REST.GraphConfig.DEFAULT:
-                        graphConfig = ConfigProperties.GRAPH_CONFIG_PROPERTY;
-                        break;
-                    case REST.GraphConfig.COMPUTER:
-                        graphConfig = ConfigProperties.GRAPH_COMPUTER_CONFIG_PROPERTY;
-                        break;
-                    default:
-                        throw new RuntimeException("Unrecognised graph config: " + graphConfig);
-                }
-            }
-            return new String(Files.readAllBytes(Paths.get(prop.getPath(graphConfig))), StandardCharsets.UTF_8);
-        } catch (IOException e) {
-            throw new GraknEngineServerException(500, ErrorMessage.NO_CONFIG_FILE.getMessage(prop.getPath(graphConfig)));
+        Properties properties = new Properties();
+        properties.putAll(GraknEngineConfig.getInstance().getProperties());
+        switch (graphConfig) {
+            case REST.GraphConfig.DEFAULT:
+                break; // Factory is already correctly set
+            case REST.GraphConfig.COMPUTER:
+                properties.setProperty(FACTORY_INTERNAL, properties.get(FACTORY_ANALYTICS).toString());
+                break;
+            default:
+                throw new RuntimeException("Unrecognised graph config: " + graphConfig);
```

Comment:
In this case we need to. This is because analytics requires one type of graph computer and titan requires another. 

There might be a more elegant way to do this though.